### PR TITLE
Entity Verification Framework

### PR DESCRIPTION
**Pull request by DRvanR**

### CODE REVIEW

**Comment by DRvanR** [comment]
@rjkip @arothuis thoughts, opinions? Shoot!

File: Vagrantfile [hunk 1]
```diff
@@ -23,15 +25,28 @@ Vagrant.configure("2") do |config|
     config.vm.hostname = "ops-dev.engine.coin.surf.net"
     config.vm.network :private_network, ip: "192.168.67.10"
 
-    config.hostsupdater.aliases = [
-            config.vm.hostname
-     ]
```
Comment (DRvanR) [inline Vagrantfile]: my bad, will put this back - guess I missed this chunk :cry:

File: app/config/parameters.yml.dist [hunk 2]
```diff
@@ -4,7 +4,7 @@
 parameters:
     database_host:     127.0.0.1
     database_port:     ~
-    database_name:     symfony
+    database_name:     conext
```
Comment (rjkip) [inline app/config/parameters.yml.dist]: Nit pick: maybe something more uniquely identifying? `conext_ops`, `conext_ops_support`

Comment (DRvanR) [inline app/config/parameters.yml.dist]: check (same with user and password)

File: src/Surfnet/Conext/EntityVerificationFramework/Api/VerificationSuiteResult.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+namespace Surfnet\Conext\EntityVerificationFramework\Api;
+
+interface VerificationSuiteResult
+{
+    /**
+     * @param VerificationTestResult $verificationTestResult
+     * @param VerificationTest       $failedTest
+     * @return VerificationSuiteResult
+     */
+    public static function failedTest(VerificationTestResult $verificationTestResult, VerificationTest $failedTest);
```
Comment (rjkip) [inline src/Surfnet/Conext/EntityVerificationFramework/Api/VerificationSuiteResult.php]: Why is this included in the interface?

Comment (DRvanR) [inline src/Surfnet/Conext/EntityVerificationFramework/Api/VerificationSuiteResult.php]: check, will remove it.

File: src/Surfnet/Conext/EntityVerificationFramework/Exception/Exceptional.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Surfnet\Conext\EntityVerificationFramework\Exception;
+
+/**
+ * Interface Exceptional
+ * @package Surfnet\Conext\EntityVerificationFramework\Exception
+ *
+ * Marker Interface
+ */
+interface Exceptional
```
Comment (rjkip) [inline src/Surfnet/Conext/EntityVerificationFramework/Exception/Exceptional.php]: Typo or intentional? Never seen an adjective as interface name.

File: src/Surfnet/Conext/EntityVerificationFramework/Repository/EntityMetadataRepositoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+namespace Surfnet\Conext\EntityVerificationFramework\Repository;
+
+use Surfnet\Conext\EntityVerificationFramework\Value\Entity;
+use Surfnet\Conext\EntityVerificationFramework\Value\EntityMetadata;
+
+interface EntityMetadataRepositoryInterface
+{
+    /**
+     * @param Entity $entity
+     * @return EntityMetadata
```
Comment (rjkip) [inline src/Surfnet/Conext/EntityVerificationFramework/Repository/EntityMetadataRepositoryInterface.php]: `EntityMetadata|null`

File: src/Surfnet/Conext/EntityVerificationFramework/TestResult.php [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+final class TestResult implements VerificationTestResult
+{
+    /**
+     * Severity Constants
+     */
+    const SEVERITY_CRITICAL = 5;
+    const SEVERITY_HIGH     = 4;
+    const SEVERITY_MEDIUM   = 3;
+    const SEVERITY_LOW      = 2;
+    const SEVERITY_TRIVIAL  = 1;
```
Comment (rjkip) [inline src/Surfnet/Conext/EntityVerificationFramework/TestResult.php]: Move up to interface

**Comment by arothuis** [comment]
:+1:
